plugins/network-elements/nuage-vsp/src/com/cloud/network/element/NuageVspElement.java

```diff
@@ -368,18 +492,15 @@ public boolean applyStaticNats(Network config, List<? extends StaticNat> rules)
             sourceNatDetails.add(sourceNatDetail);
         }
         try {
-            try {
-                HostVO nuageVspHost = getNuageVspHost(config.getPhysicalNetworkId());
-                ApplyStaticNatVspCommand cmd = new ApplyStaticNatVspCommand(networkDomain.getUuid(), vpcOrSubnetUuid, isL3Network, sourceNatDetails);
-                ApplyStaticNatVspAnswer answer = (ApplyStaticNatVspAnswer)_agentMgr.easySend(nuageVspHost.getId(), cmd);
-                if (answer == null || !answer.getResult()) {
-                    s_logger.error("ApplyStaticNatNuageVspCommand for network " + config.getUuid() + " failed");
-                    if ((null != answer) && (null != answer.getDetails())) {
-                        throw new ResourceUnavailableException(answer.getDetails(), Network.class, config.getId());
-                    }
+            HostVO nuageVspHost = getNuageVspHost(config.getPhysicalNetworkId());
+            ApplyStaticNatVspCommand.Builder cmdBuilder = new ApplyStaticNatVspCommand.Builder().networkDomainUuid(networkDomain.getUuid())
+                    .networkUuid(config.getUuid()).vpcOrSubnetUuid(vpcOrSubnetUuid).isL3Network(isL3Network).isVpc(vpcId != null).staticNatDetails(sourceNatDetails);
+            ApplyStaticNatVspAnswer answer = (ApplyStaticNatVspAnswer)_agentMgr.easySend(nuageVspHost.getId(), cmdBuilder.build());
+            if (answer == null || !answer.getResult()) {
+                s_logger.error("ApplyStaticNatNuageVspCommand for network " + config.getUuid() + " failed");
+                if ((null != answer) && (null != answer.getDetails())) {
+                    throw new ResourceUnavailableException(answer.getDetails(), Network.class, config.getId());
                 }
-            } catch (Exception e) {
-                s_logger.warn("Failed to apply static Nat in Vsp " + e.getMessage());
             }
         } catch (Exception e) {
```

Comment:
Why are all checked and unchecked exceptions being caught here?  Why not catch the subset of expected checked exceptions?